src/basic/process-util.c

```diff
@@ -675,7 +675,7 @@ int getenv_for_pid(pid_t pid, const char *field, char **_value) {
                 }
                 line[i] = 0;
 
-                if (memcmp(line, field, l) == 0 && line[l] == '=') {
+                if (strneq(line, field, l) && line[l] == '=') {
```

Comment:
Exactly. If `strneq(..., l)` returns true, we know that the layout of `line` in memory is `[field[0], field[1], ..., field[l-1], x, ...]`, with at least l+1 characters initialized, where `x` could be `0` or some character. Either way, `line[l]` is safe to access.